tests/Application/Actions/Donations/CreateTest.php

```diff
@@ -871,6 +873,46 @@ public function testSuccessWithMinimalData(): void
         $this->assertEquals('123CampaignId12345', $payloadArray['donation']['projectId']);
     }
 
+    public function testErrorWhenAllDbPersistCallsFail(): void
```

Comment:
the code you added to the Create controller is very simple and IMHO doesn't need testing. But the if condition in the DonationService probably does. I feel like we'd probably be able to write a significantly simpler and clearer test if it was a test for the DonationService, rather than a test for the Create controller. 

This test also doesn't mutation-cover the call to `$this->logger->error`, which I think is the main point of this PR. @NoelLH what do you think about testing the DonationService instead?

Reply:
https://output.circle-artifacts.com/output/job/58cf5124-bb3d-46de-8f7d-97c369b00810/artifacts/0/reports/mutation-testing-all-files/infection.html#mutant/Domain/DonationService.php 

![image](https://github.com/thebiggive/matchbot/assets/159481/4641739b-5327-4702-8c26-9530ebe9af72)

Reply:
Good point. I would definitely prefer to do both, I think it's good to also cover the end to end behaviour for such an important endpoint. Agree that the greater value is in testing `DonationService`.

Reply:
@NoelLH I think covering the end to end happy path behaviour is important, I don't really see the value in testing at this level for the error case.

Reply:
Maybe - I think these sort of tests are not all that cheap to own, they tend to slow down changes quite a lot.